### PR TITLE
Cleanup `getDataType` in XSMM and tile and fuse (NFC)

### PR DESCRIPTION
**Pull request by chelini**
- `getDataType` is now an utils. Need as utils for later patches.

- Use .td to specify dialects deps for tile and fuse.

### CODE REVIEW

File: include/TPP/Passes.td [hunk 1]
```diff
@@ -218,6 +218,8 @@ def TileConsumerAndFuseProducers : Pass<"tile-consumer-and-fuse-producers",
            "true", "Fuse from the last fusable consumer of the current target">,
     Option<"useForAll", "use-for-all", "bool", "true", "Use parallel forAll">
   ];
+  let dependentDialects = ["linalg::LinalgDialect", "scf::SCFDialect",
```
Comment (KavithaTipturMadhu) [inline include/TPP/Passes.td:221]: Is this an unrelated change?

**Review by KavithaTipturMadhu — APPROVED** [review]

File: include/TPP/Dialect/Xsmm/XsmmUtils.h [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+#include "TPP/Dialect/Xsmm/XsmmEnum.h"
+
+namespace mlir {
+class Type;
+class RewriterBase;
+
+namespace xsmm {
+namespace utils {
+
+xsmm::DataTypeAttr getXsmmDataType(RewriterBase &rewriter, Type type);
```
Comment (rengolin) [inline include/TPP/Dialect/Xsmm/XsmmUtils.h:21]: Nit: If this is already in the `xsmm` namespace, does it really need to be called `getXsmmDataType`?

**Review by rengolin — APPROVED** [review]
